runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/KingdomPlugin.java

```diff
@@ -122,20 +203,90 @@ private void removeKingdomInfobox()
 		}
 	}
 
+	private int estimateCoffer(long lastChangedEpoch, int lastCoffer)
+	{
+		int daysSince = (int) Duration.between(Instant.now(), Instant.ofEpochSecond(lastChangedEpoch)).toDays();
+		int maxDailyWithdrawal = isRoyalTroubleCompleted() ? MAX_WITHDRAWAL_ROYAL_TROUBLE : MAX_WITHDRAWAL_BASE;
+		int maxDailyThreshold = maxDailyWithdrawal * 10;
+
+		for (int i = 0; i < Math.max(daysSince, 0); i++)
+		{
+			lastCoffer -= (lastCoffer > maxDailyThreshold) ? maxDailyWithdrawal : lastCoffer / 10;
+		}
+		return lastCoffer;
+	}
+
+	private int estimateFavor(long lastChangedEpoch, int lastFavor)
+	{
+		int daysSince = (int) Duration.between(Instant.now(), Instant.ofEpochSecond(lastChangedEpoch)).toDays();
+		float dailyPercentage = isRoyalTroubleCompleted() ? FAVOR_DECREMENT_ROYAL_TROUBLE : FAVOR_DECREMENT_BASE;
+
+		lastFavor -= (int) (Math.max(daysSince, 0) * dailyPercentage * MAX_FAVOR);
+		return Math.max(lastFavor, 0);
+	}
+
 	private boolean isInKingdom()
 	{
 		return client.getLocalPlayer() != null
 			&& KINGDOM_REGION.contains(client.getLocalPlayer().getWorldLocation().getRegionID());
 	}
 
-	private boolean hasCompletedQuest()
+	private boolean isThroneOfMiscellaniaCompleted()
 	{
 		return client.getVar(VarPlayer.THRONE_OF_MISCELLANIA) > 0;
 	}
 
+	private boolean isRoyalTroubleCompleted()
+	{
+		return Quest.ROYAL_TROUBLE.getState(client) == QuestState.FINISHED;
+	}
+
 	static int getFavorPercent(int favor)
 	{
-		return (favor * 100) / 127;
+		return (favor * 100) / MAX_FAVOR;
+	}
+
+	private void sendChatMessage(String chatMessage)
+	{
+		final String message = new ChatMessageBuilder()
+			.append(ChatColorType.HIGHLIGHT)
+			.append(chatMessage)
+			.build();
+
+		chatMessageManager.queue(
+			QueuedMessage.builder()
+				.type(ChatMessageType.CONSOLE)
+				.runeLiteFormattedMessage(message)
+				.build());
 	}
 
+	protected Instant getLastChanged()
```

Comment:
These methods can all be `private` or package-private instead of `protected` as we will not subclass the plugin.
```suggestion
	private Instant getLastChanged()
```

Specifically, it looks like each of these added methods can be `private` besides `getCoffer()` and `getFavor()` as they are referenced from the counter.